esp/services/ws_machine/componentstatus.cpp

```diff
@@ -331,7 +331,7 @@ static CComponentStatusFactory *csFactory = NULL;
 
 static CriticalSection getComponentStatusSect;
 
-extern COMPONENTSTATUS_API IComponentStatusFactory* getComponentStatusFactory()
```

Comment:
Not clear why this is the right fix. Why is this function declared extern "C" in the header file?